### DISCUSSION TITLE
libssh2 1.11.1 release?

### DISCUSSION THREAD

**Discussion by vszakats** [libssh2/libssh2#1249] [Ideas]
There has been a fair amount of mostly fixes and tidy-ups since 1.11.0.

196 commits at the moment: https://github.com/libssh2/libssh2/compare/libssh2-1.11.0...f0e23dae9128185d1af4345360f0bd712d593a07

With niceties such as OpenSSL 3 `no-deprecated` compatibility, REUSE
compatibility, disablable MD5, AES-GCM-encrypted private keys, CMake
improvements, \*BSD CI tests, a few build regression fixes and more. Almost
all of these backed by new CI tests (up to 84 from 61). We also lose the
`Makefile.mk` and `NMakefile` build methods. And fix CVE-2023-48795.

It'd be real nice to include the [chacha20-poly1305@openssh.com](https://github.com/libssh2/libssh2/pull/1207#issuecomment-1821429373) patch
in this release. [PR #1426] [DONE]

And still pending #1243, which is green now and ready for review. [DONE]

After this release it came up to do an ABI break next. This would allow
to address this long-term issue: #1220 and those listed in [TODO](https://github.com/libssh2/libssh2/blob/c6589b8823e9d4a282587a6a5dab831b54d412c0/docs/TODO#L26-L64). In case
we agree to commit to this.

/cc @libssh2/dev-team

**Comment by bagder** [comment]
I'm :+1: for a release. We mostly need to get the `RELEASE-NOTES` done, then I think I can do all the mechanics necessary to ship.

**Comment by vszakats** [comment]
Release notes updates here:
https://github.com/libssh2/libssh2/blob/52c382f7aae2a8bff4ce3b4b83d18dde30ea11e2/RELEASE-NOTES

It's a fresh script + manual work. I did my best, but please let me know
if somebody or something is missing or might need improvement.

Helper script here:
https://gist.github.com/vszakats/70dbe907de571c2beaa5e793d4f868da

PR for further updates:
#1279

**Comment by vszakats** [comment]
If nobody has anything else pending, we're ready for release.

**Reply by pm-cfs** [reply]
Would be great to get the Terrapin fix in there for CVE-2023-48795 - given the timeframe between releases, addressing this major SSH issue now seems timely.

**Reply by vszakats** [reply]
We merged the fix, it will be in the release.

**Comment by ivoanjo** [comment]
+1 would be really cool to see a release out with https://github.com/libssh2/libssh2/pull/1058 :)

**Comment by willco007** [comment]
@MichaelBuckley is investigating a possible regression, please hold off until we give an all clear.

**Comment by willco007** [comment]
OK, fix has been landed. All clear.

**Comment by vszakats** [comment]
@willco007 & @MichaelBuckley: Thanks!

Any chance somebody would be kind to peer-review my #1297 and #1301?
These are also security related and it'd be nice to include them in the release,
and that could really use such review and approval to avoid any mistake there.

**Reply by MichaelBuckley** [reply]
I'll take a look in a little bit here.

**Reply by vszakats** [reply]
@MichaelBuckley Thank you for you reviews! Both patches are merged now.

**Comment by edward-shen** [comment]
Hey all! Just wondering on the status of this. Are there any remaining blockers left?  Or are we just waiting on a cut to be made?

**Reply by vszakats** [reply]
There is #1303 that sprung from the review of #1301; adding futher low-level fail checks. Nothing else.

**Reply by vszakats** [reply]
Merged now.

**Comment by vszakats** [comment]
Comments and suggestions are appreciated for deprecation
plans for obsolete algos: #1307

**Comment by tio-checo** [comment]
Hi guys,
I am wondering if there is any ETA for 1.11.1 release. In particular, I am interested in fix for CVE-2023-48795.
Thank you,
Jan

**Comment by pm-cfs** [comment]
Starting to get comments from users using ChaCha as many servers rollout their Terrapin updates.
Any way we can get a release out in the next 5-7 days?

Lots of other good stuff in here that would be nice to have too.
Thanks for all the hard work; we do appreciate it!

**Comment by miertschink** [comment]
I am still confused if the use of libssh2 makes my setup really vulnerable to [CVE-2023-48795](https://github.com/advisories/GHSA-45x7-px36-x8w8)

afaik libssh2 does not support ChaCha20-Poly1305 and also not any MAC algorithm suffixed -etm@openssh.com

but... here is part of the FAQ from terrapin-attack.com:
_In more technical terms, if your SSH implementations supports (and is configured to offer) the `chacha20-poly1305@openssh.com `encryption algorithm, or any encryption algorithm suffixed `-cbc` in combination with any MAC algorithm suffixed `-etm@openssh.com`, you are vulnerable to Terrapin._

So even if there is no release (that enables support of strict key exchange) I am still not vulnerable to this attack as libssh2 does not support this combination.

Is that correct?

**Reply by tio-checo** [reply]
Actually, libssh2.org lists hmac-sha2-512-etm@openssh.com & hmac-sha2-256-etm@openssh.com as supported MAC algorithms.

**Reply by miertschink** [reply]
ah, I see
I am using PHP with libssh2 and looked it up here: https://www.php.net/manual/en/function.ssh2-connect.php 
there is this parameter "methods" where a client_to_server or server_to_client configuration array can be passed
The "Supported Values" there doesn't mention these. 
However there is a hint that says 
_"* - Supported Values are dependent on methods supported by underlying library. See [» libssh2](http://libssh2.org/) documentation for additional information."_

Ok, basically I am vulnerable.
Thank you!

**Comment by YamsThePotato** [comment]
Hello!  Is it still planned to release a 1.11.1 release?  I'm specifically interested in getting the fix for [CVE-2023-48795](https://github.com/advisories/GHSA-45x7-px36-x8w8).

**Reply by sburnwal** [reply]
I am interested in this too CVE-2023-48795. I understand it would be released but would like to know a rough estimate of the timeline to release it.

**Comment by vszakats** [comment]
After processing the backlog and merging more improvements and fixes,
the repo is (again) fit for a release 1.11.1.

~~[ #1315 is a larger PR pending addition. It may take some more while
to merge it, and perhaps better fitting 1.12.0. ]~~ [merged]

/cc @bagder

**Reply by Tejaswikandula** [reply]
Does the release 1.11.1 including all commits from latest master. I'm interested to know if [#1314](https://github.com/libssh2/libssh2/pull/1314) would be part of 1.11.1 release

**Reply by willco007** [reply]
Yes it would.

**Reply by dmondrik** [reply]
Any update on a date when we might see the release tag created?

**Comment by Jan-E** [comment]
I am admiring all the work that @vszakats is doing the last time for LIbSSH2, but really would like to see a release as well. The Terrapin fix was already merged in December. Is anything blocking the 1.11.1 release?

**Comment by vszakats** [comment]
To make the wait more useful for everyone, we appreciate testing and feedback
on current master (as a "release candidate") while a release is cut.

Also for somebody interested in mbedTLS, we have an unfixed regression with
mbedTLS 3.6.0: 2e4c5ec4627b3ecf4b6da16f365c011dec9a31b4
https://github.com/libssh2/libssh2/blob/3b23e039f74a637b572213fb20c48673708ae228/src/mbedtls.c#L1295-L1315

I was thinking to make it a compile-time failure outside of CI, to avoid 
runtime surprises. That's the only pending thing I have. But a proper fix of
course would be much better. → #1393

**Comment by vszakats** [comment]
We also have this unfixed regression with wolfSSL, since 1.11.0:
#1020, #1299

**Reply by vszakats** [reply]
#1020 and #1299 fixed → #1408, #1409, #1410, #1411, #1407

**Comment by Jan-E** [comment]
> @vszakats closed https://github.com/libssh2/libssh2/discussions/1249 as resolved.

Resolved? Will there be a 1.11.1 release soon?

**Comment by vszakats** [comment]
The other options are "duplicate" or "outdated". None is fitting. A release would be nice, but I'm abandoning my personal push for one. Tried my best.

**Comment by Jan-E** [comment]
OK, clear. Thanks for all the work you are still doing for LibSSH2.

**Comment by willco007** [comment]
@bagder @vszakats  Reopening this issue, we need to get a release out. What can we do to make this happen?

**Comment by bagder** [comment]
I'm willing to do the actual release tarball and upload if that helps. But I'm not really involved in the project right now so someone would need to make sure to populate `RELEASE-NOTES` appropriately and whatever else that needs to be done to make a release shine.

Just tell me when things are in shape and I'll package it up.

**Reply by willco007** [reply]
It's ready to go!

**Comment by willco007** [comment]
@vszakats had the release notes updated at one point, but there needs to be more issues added at this point. Is that something you could do again @vszakats?  Thanks!

**Comment by vszakats** [comment]
I've updated the release notes and pushed a few pending updates. Also repaired AppVeyor CI to the degree that it's now green, but with tests disabled in many jobs.

It'd be nice if everyone interested could make a local test build and report if there is anything to fix before release. Thanks!

**Comment by vszakats** [comment]
The repo is ready for a release.

**Comment by bagder** [comment]
Done!
